apps/extension/src/ui/hooks/useAddressBook.tsx

```diff
@@ -1,36 +1,34 @@
+import { HexString } from "extension-shared"
 import { useCallback } from "react"
-import { useTranslation } from "react-i18next"
 
-import { AddressBookContact, addressBookStore } from "@extension/core"
+import { api } from "@ui/api"
 import { useContacts } from "@ui/state"
 
+type ContactInfo = { address: string; name: string; genesisHash?: HexString }
+
 export const useAddressBook = () => {
-  const { t } = useTranslation()
   const contacts = useContacts()
 
-  const add = useCallback(
-    ({ address, ...rest }: AddressBookContact) =>
-      addressBookStore.set({ [address]: { address, ...rest } }),
-    [],
-  )
+  const add = useCallback(({ address, name, genesisHash }: ContactInfo) => {
+    api.accountAddExternal([
+      {
+        type: "contact",
+        name,
+        address,
+        genesisHash,
+      },
+    ])
+  }, [])
```

Comment:
We call `await add()` inside of `apps/extension/src/ui/domains/Settings/AddressBook/ContactCreateModal.tsx`, but we're not currently returning the Promise here 

```suggestion
  const add = useCallback(
    ({ address, name, genesisHash }: ContactInfo) =>
      api.accountAddExternal([
        {
          type: "contact",
          name,
          address,
          genesisHash,
        },
      ]),
    [],
```